hexrd/instrument.py

```diff
@@ -3573,7 +3580,245 @@ def simulate_laue_pattern(self, crystal_data,
             pass    # close loop on grains
         return xy_det, hkls_in, angles, dspacing, energy
 
+class CylindricalDetector(PlanarDetector):
+    """Base class for 2D cylindrical detector
+       
+       A cylindrical detector is a simple rectangular 
+       row-column detector which has been bent in the 
+       shape of a cylinder. Inherting the PlanarDetector 
+       class except for a few changes to account for the
+       cylindder ray intersection.
+    """
+    def __init__(self,
+                 rows=240, cols=1525,
+                 pixel_size=(0.1, 0.1),
+                 tvec=np.r_[0., 0., -49.5],
+                 tilt=ct.zeros_3,
+                 name='rind',
+                 bvec=ct.beam_vec,
+                 xrs_dist=None,
+                 evec=ct.eta_vec,
+                 saturation_level=None,
+                 panel_buffer=None,
+                 tth_distortion=None,
+                 roi=None,
+                 distortion=None,
+                 max_workers=max_workers_DFLT):
```

Comment:
I think this is a good first pass for the initial implementation to get it working, but we will probably want to make a base `Detector` class that both the `CylindricalDetector` and `PlanarDetector` inherit from.

Inheritance carries an `is-a` relationship. As in, if `CylindricalDetector` inherits from `PlanarDetector`, that means `CylindricalDetector` "is a" `PlanarDetector`, which isn't the case. However, `CylindricalDetector` "is a" `Detector`.

Then if we have a base `Detector` class, we can have the `__init__` args for `CylindricalDetector` be like:

```python
def __init__(self, cylindrical_arg1, cylindrical_arg2, **detector_kwargs):
    super().__init__(**detector_kwargs)
    ...

```
And the arguments and logic for initializing the detector would be the same for all detectors and be done in the `Detector.__init__()` method.